Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -964,7 +989,12 @@ void doVayDepositionShapeN (const GetParticlePosition& GetPosition,
         }
 #endif
     } );
-    amrex::The_Managed_Arena()->free(cost_real);
+#   if defined(WARPX_USE_GPUCLOCK)
+    if( load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::GpuClock) {
+        amrex::Gpu::synchronize();
```

Comment:
Not sure if stream synchronize would be sufficient?

Reply:
I think we are lacking an add to the WarpX `cost` array here as well, between the synchronize and the free